plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -81,6 +81,7 @@ export type KubernetesProxyOptions = {
   logger: Logger;
   clusterSupplier: KubernetesClustersSupplier;
   authStrategy: AuthenticationStrategy;
+  httpAuth: HttpAuthService;
```

Comment:
Ugh, this needs to be optional with a fallback to `createLegacyAuthAdapters` which then forces us to depend on the `DiscoveryService`. Callers of the constructor are coming from the old backend system and won't have a `HttpAuthService` instance.